### ISSUE TITLE
Compute Atomic Convolutions only on Contact Regions

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1827]
The atomic convolutions currently take in an entire protein-ligand complex. This makes working with them very gnarly since proteins get large (tens of thousands of atoms easily). Since the coordinates of the entire protein are handled, the tensors we need to handle in GPU memory get really large really quickly. This means that in practice, training an atomic convolution isn't really feasible unless you pre-restrict to binding pockets. This is feasible on PDBBind where binding pocket PDBs are available, but isn't easy for general structures.

I've been experimenting with a few different ideas for improving this situation and think I've come up with a fairly simple hack. Start by computing the pairwise distance matrix between the protein and the ligand. Prune out all protein atoms which aren't within some contact radius of the ligand (say 4.5 Angstroms, but this can be set by the user). This drastically reduces the number of atoms which need to be handled in featurization numpy arrays and tensorflow tensors. The advantage of this hack is that we no longer need to manually specify a binding pocket since the contact region implicitly delineates a contact region. We also can handle much larger proteins since the only computation that considers the fully protein is the pairwise distance computation which can be done with `scipy`

I've implemented this within `AtomicConvFeaturizer` in #1792. I'm still experimenting with it. So far featurization seems to be a good bit faster in addition to saving memory.

**Comment by vsomnath** [comment]
Is this a more general version of the `load_binding_pocket` toggle that is available within `dc.molnet.load_pdbbind`?

**Comment by rbharath** [comment]
Yep, that's the idea. The `load_binding_pocket` toggle depends on the pre-computed binding pocket PDB files that are part of PDBBind while this technique can work for any PDBs. Also, it's possible (though I haven't done this yet), that we can use this technique to model protein-protein interactions. Usually this is really hard since the complexes are too big to easily handle in memory
